### DISCUSSION TITLE
Public facing API?

### DISCUSSION THREAD

**Discussion by tefkah** [knowledgefutures/pubpub#1742] [Ideas]
Are there any plans to have a public-facing API for PubPub, or is there already one by any chance? (I looked pretty hard but couldn't find any mention of one) 

Currently, as you are still working the editorial workflow in PubPub, a lot of our editing work is being done outside of PubPub in places like OJS, which leads to the process of publishing a paper becoming one of of copy-pasting a lot of data from one CMS to another, which is both error-prone and kind of boring 😅. Ideally for me, there'd be a relatively simple API endpoint where I could send a POST request with the relevant metadata to create an article and possibly a link to a .docx/.tex/.xml file which should be turned into the content. Being able to request data from PubPub would also be nice, as would be the ability to add webhooks for when an Pub is published/modified/created etc, but making the publishing process easier is currently my biggest concern.

I of course totally understand that this is not a highly requested feature and won't be needed by like 95% of the communities and that you might not have the time to implement this! 
If it would be helpful, I'd love to help out in creating this feature! The PubPub codebase is quite large so I could not easily figure out how a Pub is created and whether that would require major rehauling in order to accommodate this, but if it is more a matter of manpower rather than architecture preventing such a thing from being created I would be more than happy to lend my help!

As always, thanks for all your work! PubPub is truly lovely.

**Comment by metasj** [comment]
I'd love this too :)  I know there are bulk-import processes, it would be great to have access to something like this from another publishing framework (as you say) or a lab notebook or personal outliner -- both to avoid cut-and-paste mistakes and to capture that source connection in an appropriate bit of metadata [in a docmap? or just in an edit/import summary]

**Reply by advaitathreya** [reply]
Is there any documentation for a self-managed bulk-import? I'm working with a student-run journal to move to Pubpub, and we have around 75 or so articles from the past that we would like to move. I was wondering if there was any way we could automate this- I can explore the API client by @tefkah but I wanted to check if there was an official way to do it since you reference it here- thanks!

**Comment by daaronr** [comment]
Agree. For our preprint review work, the process of uploading 4-5 separate files, inputting author names, adjusting their settings, embedding media, adding connections (to each other and to the original paper), and requesting DOIs for each takes several hours ATM. I'm really interested in a way to streamline this!

**Reply by tefkah** [reply]
Hi! I actually just released a client to interact with the API [here](https://github.com/tefkah/pubpub-client)! It's still quite experimental and possibly buggy, but let me know what you think and what you would want it to do!
I haven't implemented DOI minting yet but shouldn't be too hard to implement.

**Comment by tefkah** [comment]
Hi to everyone following this! I actually just released a client to interact with the API [here](https://github.com/tefkah/pubpub-client)! It's still quite experimental and possibly buggy, but let me know what you think and what you would want it to do!

**Comment by mattakamatsu** [comment]
I'm very interested in this! We want to connect our pages written in e.g. Obsidan/Notion/Roam Research, via https://samepage.network/ (which is also still very experimental) with [dvargas92495](https://github.com/dvargas92495/dvargas92495), for similar reasons that you mention above.  The promise of using Samepage is that you could have continual synching of the page between different content editors, and saved in a decentralized, verifiable format via [IPFS](https://medium.com/@luishrsoares/ipfs-the-interplanetary-file-system-explained-9aeeab56484c).

**Comment by taniyuki2** [comment]
there'd be a relatively simple API endpoint where I could send a POST request with the relevant metadata to create an article and possibly a link to a .docx/.tex/.xml file which should be turned into the content.
[tiny fishing]( https://tinyfishing.co)
